### ISSUE TITLE
Automatic schedule generation

### ISSUE THREAD

**Issue by brennanmcmicking** [brennanmcmicking/scheduler#30]
The scheduler should have some way to generate a list of all valid schedules, even if this is a separate view from the normal course-selector view.

My idea:
1. user creates a schedule, gets taken to `/schedule/:schedule_id`
2. user adds some courses
3. user clicks "generate valid schedules", taking them to a read-only view at `/schedule/:schedule_id/valid`
4. user can flip through all of the valid schedules
5. user can click "take" or "apply" on a valid schedule and get taken back to `/schedule/:schedule_id` with the new section selections applied
6. user can continue hand-editing the schedule

**Comment by malcolmseyd** [comment]
Do you think we're ready for me to work on this feature? If so I'd be happy to whip up a page.

I'm assuming I can just build the view as a fullscreen thingy and we can tweak it later. Make it a popup view or something with HTMX or an iframe.

**Comment by brennanmcmicking** [comment]
Yeah go for it! I think it's fine to stay as a fullscreen page where the user can add courses and then flip through valid schedules, then "take" them to save them as a schedule in their cookie jar and start hand-editing them
